tests/qos/test_qos_sai.py

```diff
@@ -181,6 +181,63 @@ def testQosSaiHeadroomPoolSize(self, ptfhost, dutTestParams, dutConfig, dutQosCo
         testParams.update(dutTestParams["basicParams"])
         self.runPtfTest(ptfhost, testCase="sai_qos_tests.HdrmPoolSizeTest", testParams=testParams)
 
+    @pytest.mark.parametrize("bufPool", ["wm_buf_pool_lossless", "wm_buf_pool_lossy"])
+    def testQosSaiBufferPoolWatermark(self, request, bufPool, ptfhost, dutTestParams, dutConfig, dutQosConfig, ingressLosslessProfile, egressLossyProfile, resetWatermark):
+        """
+            Test QoS SAI Queue buffer pool watermark for lossless/lossy traffic
+
+            Args:
+                ptfhost (AnsibleHost): Packet Test Framework (PTF)
+                dutTestParams (Fixture, dict): DUT host test params
+                dutConfig (Fixture, dict): Map of DUT config containing dut interfaces, test port IDs, test port IPs,
+                    and test ports
+                dutQosConfig (Fixture, dict): Map containing DUT host QoS configuration
+                ingressLosslessProfile (Fixture): Map of ingress lossless buffer profile attributes
+                egressLossyProfile (Fixture): Map of egress lossy buffer profile attributes
+                resetWatermark (Fixture): reset watermarks
+
+            Returns:
+                None
+
+            Raises:
+                RunAnsibleModuleFail if ptf test fails
+        """
+        disableTest = request.config.getoption("--disable_test")
+        if disableTest:
+            pytest.skip("Buffer Pool watermark test is disabled")
+
+        portSpeedCableLength = dutQosConfig["portSpeedCableLength"]
+        if "wm_buf_pool_lossless" in bufPool:
+            qosConfig = dutQosConfig["param"][portSpeedCableLength]
+            triggerDrop = qosConfig[bufPool]["pkts_num_trig_pfc"]
+            fillMin = qosConfig[bufPool]["pkts_num_fill_ingr_min"]
+            buf_pool_roid = ingressLosslessProfile["bufferPoolRoid"]
```

Comment:
nit: mix of camel case and hyphenated naming

Reply:
Agreed. will clean up this entire script later.